File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.py
from collections import deque

def solution(operations):
    answer = []
    queue = []
    top = []
    command = deque()
    for oper in operations :
        code, num = oper.split()
        num = int(num)
        command.append((code, num))
    while command : 
        code, num = command.popleft()
        if code == "I" :
            top.append(num)
        if top:
            if code == "D" and num < 0 :
                minNum =min(top)
                top.remove(minNum)
            elif code == "D" and num > 0 :
                maxNum = max(top)
                top.remove(maxNum)
        else :
            continue
    if top :
        nmin = min(top)
        nmax = max(top)
        return [nmax, nmin]
        
    else :
        return [0, 0]
            
            
        
    
        
    return answer

File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.js
function solution(operations) {
  const top = [];

  for (const oper of operations) {
    const [code, numStr] = oper.split(" ");
    const num = Number(numStr);

    if (code === "I") {
      top.push(num);
    } else if (code === "D") {
      if (top.length === 0) continue;

      if (num < 0) {
        const minNum = Math.min(...top);
        const index = top.indexOf(minNum);
        top.splice(index, 1);
      } else if (num > 0) {
        const maxNum = Math.max(...top);
        const index = top.indexOf(maxNum);
        top.splice(index, 1);
      }
    }
  }

  if (top.length === 0) {
    return [0, 0];
  }

  return [Math.max(...top), Math.min(...top)];
}